src/Orleans/Statistics/AverageTimeSpanStatistic.cs

```diff
@@ -153,9 +153,9 @@ public override string ToString()
             return GetValueString();
         }
 
-        public void TrackMetric(Logger logger)
+        public void TrackMetric(ITelemetryClient telemetryClient)
         {
-            logger.TrackMetric(currentName, GetCurrentValue());
+            telemetryClient.TrackMetric(currentName, GetCurrentValue());
```

Comment:
I'm not sure ITelemetryClient is a good name. Maybe TelemetryProducer? Since this is a public interface. I'd like more people's opinion. 
What do you think ? @sergeybykov @jason-bragg .